### PR TITLE
[WMV-1244] Added a Map component to Ripple.

### PR DESCRIPTION
**Pull request by danlaush**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/WMV-1244

### Changed

1.  Added a map component, initially for displaying the #VicFreeWifi map but built in such a way that future map layers will be easily configurable.

### Screenshots

screen shot 2018-12-17 at 4 25 17 pm

screen shot 2018-12-17 at 4 25 29 pm

### CODE REVIEW

File: packages/Molecules/Map/MapIndicator.vue [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    &__close {
+      position: absolute;
+      right: .75rem;
+      top: .75rem;
+    }
+
+    &__title {
+      color: blue;
+      font-size: rem-calc(13);
```
Comment (tim-yao) [inline packages/Molecules/Map/MapIndicator.vue]: Please use `rpl-font-size()` to set all font size. A list of existing font size https://github.com/dpc-sdp/ripple/blob/7d80c68558a70b4198ce2527d6d00359fa08544d/packages/Atoms/Global/scss/_font.scss#L3
We don't have 13px, so please use the closest size `rpl-font-size(xs)` which is `14px`.

File: packages/Molecules/Map/MapIndicator.vue [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    &__close {
+      position: absolute;
+      right: .75rem;
+      top: .75rem;
+    }
+
+    &__title {
+      color: blue;
```
Comment (tim-yao) [inline packages/Molecules/Map/MapIndicator.vue]: Please use `rpl-color()`, we don't have `blue` in the color palette, so maybe chose `primary`.

File: packages/Molecules/Map/MapIndicator.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.20);
+    padding: 1.25rem 1rem;
+
+    &:before {
+      $size: 12px;
+      content: '';
+      display: block;
+      width: $size;
+      height: $size;
+      background-color: white;
```
Comment (tim-yao) [inline packages/Molecules/Map/MapIndicator.vue]: Please review all color and use our `rpl-color`

File: packages/Molecules/Map/CloseButton.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  background: transparent;
+  line-height: 1;
+  cursor: auto;
+
+  &:hover {
+    color: grey;
+  }
+
+  &.large {
+    width: 30px;
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/CloseButton.vue]: Handled by `RplIcon` now

File: packages/Molecules/Map/CloseButton.vue [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    height: 30px;
+  }
+
+  &.small {
+    width: 20px;
+    height: 20px;
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/CloseButton.vue]: Handled by `RplIcon` now

File: packages/Molecules/Map/CloseButton.vue [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  &.light {
+    color: white;
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/CloseButton.vue]: Handled by `RplIcon` now

File: packages/Molecules/Map/Map.vue [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+// height: 0 means no map at all. By setting height: 0 and
+// padding-top: (9/16)% on .rpl-map__container, the map
+// itself can be height auto which OL picks up correctly.
+
+$rpl-map-aspect-ratio: (
+  xs: (8/10) * 100%,
+  s: (9/16) * 100%
+);
+
+$rpl-map-popup-width: 300px;
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/Map.vue]: 👍

File: packages/Molecules/Map/Map.vue [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+    @each $bp, $val in $rpl-map-aspect-ratio {
+      @include rpl_breakpoint($bp) {
+        padding-top: $val;
+      }
+    }
+  }
+
+  &__popup {
+    position: absolute;
+    z-index: 999;
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/Map.vue]: 👍

File: packages/Molecules/Map/MapIndicator.vue [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+      // if the selectedFeature has been set to null, then hide the popup
+      this.show = !!newFeature
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+
+  // @import '~assets/styles/includes';
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/MapIndicator.vue]: 👍

File: packages/Molecules/Map/MapIndicator.vue [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+  .standard-button {
+    // @include button($background: palette(blue,tint5), $background-hover: palette(blue,tint10));
+    border-radius: 5px;
+    font-size: rem-calc(13);
+    display: block;
+    margin-bottom: 0;
+    font-weight: 700;
```
[the remark this answers is not in the document]
Comment (danlaush) [reply packages/Molecules/Map/MapIndicator.vue]: Not actually used, removed

File: packages/Molecules/Map/MapIndicator.vue (hunk 3 above)
Comment (danlaush) [reply packages/Molecules/Map/MapIndicator.vue]: 👍

File: packages/Molecules/Map/MapIndicator.vue (hunk 2 above)
Comment (danlaush) [reply packages/Molecules/Map/MapIndicator.vue]: 👍

File: packages/Molecules/Map/MapIndicator.vue (hunk 1 above)
Comment (danlaush) [reply packages/Molecules/Map/MapIndicator.vue]: 👍

**Comment by tim-yao** [comment]
Further work will be done in https://digital-engagement.atlassian.net/browse/SDPA-2309

**Pull request merged**